pyuvdata/utils.py

```diff
@@ -492,6 +492,83 @@ def uvcalibrate(uvdata, uvcal, inplace=True, prop_flags=True, flag_missing=True,
         return uvdata
 
 
+def apply_uvflag(uvd, uvf, inplace=True, force_pol=True, unflag_first=False,
+                 flag_missing=True):
+    """
+    Apply flags from a UVFlag to a UVData instantiation. This edits inplace by default.
+
+    Parameters
+    ----------
+    uvd : UVData object
+        UVData object to add flags to.
+    uvf : UVFlag object
+        A UVFlag object in flag mode.
+    inplace : bool
+        If True overwrite flags in uvd, otherwise return new object
+    force_pol : bool
+        If True, broadcast flags to all polarizations if they do not match.
+        Only works if uvf.Npols == 1.
+    unflag_first : bool
+        If True, completely unflag the UVData before applying flags.
+    flag_missing : bool
+        If input uvf is a baseline type and antpairs in uvd do not exist in uvf,
+        flag them in uvd. Otherwise leave them untouched.
+
+    Returns
+    -------
+    UVData
+        If not inplace, returns new UVData object with flags applied
+    """
+    # assertions
+    assert uvf.mode == 'flag'
+
+    if not inplace:
+        uvd = copy.deepcopy(uvd)
+
+    # make a deepcopy by default b/c it is generally edited inplace downstream
+    uvf = copy.deepcopy(uvf)
+
+    # convert to baseline type
+    if uvf.type != 'baseline':
+        # edits inplace
+        uvf.to_baseline(uvd, force_pol=force_pol)
+
+    else:
+        # make sure polarizations match or force_pol
+        uvd_pols, uvf_pols = uvd.polarization_array.tolist(), uvf.polarization_array.tolist()
+        if set(uvd_pols) != set(uvf_pols):
+            if uvf.Npols == 1 and force_pol:
+                # if uvf is 1pol we can make them match: also edits inplace
+                uvf.polarization_array = np.repeat(uvf.polarization_array, uvd.Npols, axis=-1)
+                uvf.Npols = len(uvf.polarization_array)
+                uvf_pols = uvf.polarization_array.tolist()
+
+            else:
+                raise ValueError("Input uvf and uvd polarizations do not match")
+
+        # make sure polarization ordering is correct: also edits inplace
+        uvf.polarization_array = uvf.polarization_array[[uvd_pols.index(pol) for pol in uvf_pols]]
+
```

Comment:
There should be additional checks to be sure the time and frequency axes are the same. Ideally if they don't match, you can still flag what overlap there is, and have default behavior for anything else. But I think it's reasonable for a first attempt to simply check that the time array and freq arrays are equivalent and throw an error if they aren't.